aries_cloudagent/resolver/base.py

```diff
@@ -58,24 +60,59 @@ def native(self):
         return self.type == ResolverType.NATIVE
 
     @property
-    @abstractmethod
     def supported_methods(self) -> Sequence[str]:
-        """Return list of DID methods supported by this resolver."""
+        """Return supported methods.
+
+        DEPRECATED: Use supported_did_regex instead.
+        """
+        return []
+
+    @property
+    def supported_did_regex(self) -> Pattern:
+        """Supported DID regex for matching this resolver to DIDs it can resolve.
+
+        Override this property with a class var or similar to use regex
+        matching on DIDs to determine if this resolver supports a given DID.
+        """
+        raise NotImplementedError(
+            "supported_did_regex must be overriden by subclasses of BaseResolver "
+            "to use default supports method"
+        )
 
-    def supports(self, method: str) -> bool:
-        """Return if this resolver supports the given method."""
-        return method in self.supported_methods
+    async def supports(self, profile: Profile, did: str) -> bool:
+        """Return if this resolver supports the given DID.
+
+        Override this method to determine if this resolver supports a DID based
+        on information other than just a regular expression; i.e. check a value
+        in storage, query a resolver connection record, etc.
+        """
+        try:
+            supported_did_regex = self.supported_did_regex
+        except NotImplementedError as error:
+            if not self.supported_methods:
+                raise error
+            warnings.warn(
+                "BaseResolver.supported_methods is deprecated; "
+                "use supported_did_regex instead",
+                DeprecationWarning,
+            )
+
+            supported_did_regex = re.compile(
+                "^did:(?:{}):.*$".format("|".join(self.supported_methods))
+            )
+
+        return bool(supported_did_regex.match(did))
 
     async def resolve(self, profile: Profile, did: Union[str, DID]) -> DIDDocument:
         """Resolve a DID using this resolver."""
         py_did = DID(did) if isinstance(did, str) else did
```

Comment:
The coercing to `DID` in this case is just making sure the DID is valid. We can do this check without assigning the value as a DID to a variable:

```python
if isinstance(did, DID):
    did = str(did)
else:
    DID.validate(did)  # <- will raise an error if invalid
```

Reply:
we use py_did in the error report to get the method. 
```
            raise DIDMethodNotSupported(
                f"{self.__class__.__name__} does not support DID method {py_did.method}"
            )
```